test/testFortran.f90

```diff
@@ -60,23 +60,52 @@ program main
   allocate(yArray(nverts))
   allocate(zArray(nverts))
 
+  ! MPMesh structure
   call polympo_getMPPositions(mpMesh, numCompsCoords, numMPs, c_loc(MPPositions))
   do i = 1,numMPs 
     call assert(abs(MPPositions(3,i) - 1.1) .lt. test_epsilon, "Assert zPositions for MP array Fail")
   end do
   call polympo_getMeshNumVtxs(mpMesh, nvertsGet)
   call assert(nverts.eq.nvertsGet,"num. verts mismatch")
-
   call polympo_getMeshNumElms(mpMesh, nElmsGet)
   call assert(numElms.eq.nElmsGet,"num. elms mismatch")
+
+  ! set MP Fields
+  do i = 1,numCompsVel
+    do j = 1,numMPs 
+        MParray(i,j) = (i-1)*numCompsVel + j
+    end do
+  end do
+  call polympo_setMPVel(mpMesh, numCompsVel, numMPs, c_loc(MParray))
+  
+  ! check MP Fields
+  MParray = 1
+  call polympo_getMPVel(mpMesh, numCompsVel, numMPs, c_loc(MParray))
+  do i = 1,numCompsVel
+    do j = 1,numMPs 
+        call assert((MParray(i,j) .eq. (i-1)*numCompsVel+j), "Assert MPVel Fail")
+    end do
+  end do
+  
+
+  ! set mesh Fields
   do i = 1,numCompsVel
     do j = 1,nverts 
         Mesharray(i,j) = (i-1)*numCompsVel + j
     end do
   end do
+  call polympo_setMeshVel(mpMesh, numCompsVel, nverts, c_loc(Mesharray))
   call polympo_setMeshOnSurfVeloIncr(mpMesh, numCompsVel, nverts, c_loc(Mesharray))
   call polympo_setMeshOnSurfDispIncr(mpMesh, numCompsVel, nverts, c_loc(Mesharray))
 
+  ! check mesh Fields
+  Mesharray = 1
```

Comment:
change to `Mesharray=-1`

Reply:
https://github.com/SCOREC/polyMPO/pull/51/commits/c087b88465e69ce6fa134325934b28f08c288945